sdk/storage/Azure.Storage.Common/src/Shared/StorageSharedKeyPipelinePolicy.cs

```diff
@@ -99,18 +101,24 @@ private static void BuildCanonicalizedHeaders(StringBuilder stringBuilder, HttpM
         {
             // Grab all the "x-ms-*" headers, trim whitespace, lowercase, sort,
             // and combine them with their values (separated by a colon).
-            foreach (var header in
-                message.Request.Headers
-                .Where(static h => h.Name.StartsWith(Constants.HeaderNames.XMsPrefix, StringComparison.OrdinalIgnoreCase))
-#pragma warning disable CA1308 // Normalize strings to uppercase
-                .Select(static h => (h.Name.ToLowerInvariant(), h.Value))
-#pragma warning restore CA1308 // Normalize strings to uppercase
-                .OrderBy(static h => h.Item1.Trim()))
+            var headers = new List<HttpHeader>();
+            foreach (var header in message.Request.Headers)
+            {
+                if (header.Name.StartsWith(Constants.HeaderNames.XMsPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    headers.Add(header);
+                }
+            }
+
+            headers.Sort(static (x, y) => string.CompareOrdinal(x.Name, y.Name));
```

Comment:
Should we call Trim here to maintain existing behavior?

Reply:
No, we don't. All headers start with `Constants.HeaderNames.XMsPrefix`, so `Trim` is essentially `TrimEnd`, which doesn't affect sorting.

Reply:
In that case, we should update the [comment](https://github.com/Azure/azure-sdk-for-net/pull/31793/files/cc2e010d0f1f60932eed969bd71467c0420c9722#diff-2ca4e8d05dd448866d56400a1db672172908ed1d538cb52e4d177690ba9df139R102) that explains we trim whitespace.